app/components/user-cards.js

```diff
@@ -1,40 +1,87 @@
 import Component from '@ember/component';
-import { computed } from '@ember/object';
+import EmberObject, { computed } from '@ember/object';
+
 
 export default Component.extend({
 
     selected_option:null,
+    sort_option:null,
+    sort_order:null,
+    isSelected:null,
+    isSelectedOrder:null,
+    sortfieldList:null,
+    sortorderlist:null,
     
-    teamslist : computed ('selected_option',function(){
-        if(this.selected_option!='All Employees')
-            return this.get('members').filterBy('team',this.get('selected_option'))
-        else
-            return this.get('members').filterBy('team')
-        
+    teamslist : computed ('selected_option','sort_option','sort_order',function(){
+    
+        if(this.sort_order==='Ascending'){
+            if(this.selected_option!='All Employees')
+            
+                return this.get('members').filterBy('team',this.get('selected_option')).sortBy(this.get('sort_option'))
+            else
+                return this.get('members').filterBy('team').sortBy(this.get('sort_option'))
+        }
+        else {
+            if(this.selected_option!='All Employees')
+                return this.get('members').filterBy('team',this.get('selected_option')).sortBy(this.get('sort_option')).reverse()
+            else
+                return this.get('members').filterBy('team').sortBy(this.get('sort_option')).reverse()
+        }
     }) ,
 
     teams: computed ('members.[]',function(){
-        
-         let teamNames =  this.members.mapBy('team').uniq();
-         teamNames.push('All Employees');
-         return teamNames;
-        
+        let teamNames =  this.members.mapBy('team').uniq();
+        teamNames.push('All Employees');
+        return teamNames;
+       
+    }),
+
+    sortfieldArrayList: computed('sortfieldList.[]',function(){
+        return this.sortfieldList.filterBy('title')
+    }),
+
+    sortorderArrayList:computed('sortorderlist.[]',function(){
+        return this.sortorderlist.filterBy('title')
     }),
     
     init() {
         this._super(...arguments);
         this.set('selected_option','All Employees')
+
+        this.set('sortfieldList', [
+            EmberObject.create({ field: 'first_name',title:'First Name' }),
+            EmberObject.create({ field: 'last_name',title:'Last Name' }),
+            EmberObject.create({ field: 'joiningDate',title:'Joining Date' })
+            
+          ]);
+
+        this.set('sortorderlist', [
+            EmberObject.create({ title: 'Ascending' }),
+            EmberObject.create({ title: 'Descending' }),
+          ]);
+
+        this.set('sort_option','first_name')
+        this.set('sort_order','Ascending')
+        this.set('isSelected','first_name');
+        this.set('isSelectedOrder','Ascending')
+
```

Comment:
We are declaring redundant properties here. Should be possible to do it with one set of properties.

And, we generally use variable names starting with 'be verbs' like is, has, will, does, did etc for `boolean` type variables.

Reply:
removed the redundant properties.